src/main/java/seedu/address/logic/commands/ScheduleCommand.java

```diff
@@ -77,6 +80,10 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException(MESSAGE_CONFLICTING_INTERVIEW);
         }
 
+        if (!model.hasMatchingAvailability(toAdd)) {
+            throw new CommandException(MESSAGE_CANDIDATE_NOT_AVAILABLE);
+        }
```

Comment:
Nice work Charmaine! Just a food for thought to consider, is it necessary to go through the entire Model/InterviewSchedule/UniqueInterviewList if we are not checking against other interviews!

Reply:
Thanks for reviewing the implementation! :) I relooked at it, and I agree, I think the layers of abstraction was unnecessary, I've instead made the implementation for the methods within the `Interview` class. Let me finalise the additional commits before re-requesting for approval!